### PR TITLE
Run the `go_e2e_deps` step in the flakes finder pipeline

### PR DESCRIPTION
**Pull request by FlorentClarret**
### What does this PR do?

Run the `go_e2e_deps` step in the flakes finder pipeline

### Motivation

- The `go_e2e_deps` was [added recently](https://github.com/DataDog/datadog-agent/pull/27802) and is now needed by the e2e test jobs
- The flakes finder pipeline is [broken](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/40184352) because we get rid of all the `needs` steps, which includes this new one
- Adding this job in this pipeline and specify the tests need it

### CODE REVIEW

File: tasks/testwasher.py [hunk 1]
```diff
@@ -229,6 +224,7 @@ def generate_flake_finder_pipeline(ctx, n=3):
                 ):
                     new_job['variables']['E2E_COMMIT_SHA'] = "$PARENT_COMMIT_SHA"
             new_job["rules"] = [{"when": "always"}]
+            new_job["needs"] = ["!reference [.needs_new_e2e_template]"]
```
Comment (FlorentClarret) [inline tasks/testwasher.py:227]: I have to have the dep job out of the `flake-finder-{i}`, fixing it

Comment (CelianR) [inline tasks/testwasher.py:227]: **❓ question:** ‏Does this work ? This is an array containing the string `.needs_new_e2e_template` tagged with `reference` != just a string

Comment (FlorentClarret) [reply tasks/testwasher.py:227]: Isn't `needs` an array? https://docs.gitlab.com/ee/ci/yaml/#needs

**Review by pducolin — APPROVED** [review]

File: tasks/testwasher.py (hunk 1 above)
Comment (FlorentClarret) [reply tasks/testwasher.py]: discussed offline! We changed the way to do it

**Review by CelianR — APPROVED** [review]

**Pull request merged**